src/utils/vc.js

```diff
@@ -65,31 +68,17 @@ export function getSuiteFromKeyDoc(keyDoc) {
   });
 }
 
-/**
- * Check if credential has Dock specific revocation
- * @param credential
- * @returns {Boolean}
- */
-export function hasDockRevocation(credential) {
-  return credential.credentialStatus
-    && (credential.credentialStatus.type === RevRegType)
-    && credential.credentialStatus.id.startsWith(DockRevRegQualifier)
-    && isHexWithGivenByteSize(credential.credentialStatus.id.slice(DockRevRegQualifier.length), RevRegIdByteSize);
-}
-
 /**
  * Checks if the revocation check is needed. Will return true if `forceRevocationCheck` is true else will check the
  * truthyness of revocationApi. Will return true even if revocationApi is an empty object.
- * @param {object} credStatus - The `credentialStatus` field in a credential. Does not care about the correct
- * structure of this field but only the truthyness of this field. The intention is to check whether the credential h
- * had a `credentialStatus` field.
+ * @param {object} credential - The credential, expecing a `credentialStatus` field
  * @param {boolean} forceRevocationCheck - Whether to force the revocation check.
  * Warning, setting forceRevocationCheck to false can allow false positives when verifying revocable credentials.
  * @param {object} revocationApi - See above verification methods for details on this parameter
  * @returns {boolean} - Whether to check for revocation or not.
  */
-export function isRevocationCheckNeeded(credStatus, forceRevocationCheck, revocationApi) {
-  return !!credStatus && (forceRevocationCheck || !!revocationApi);
+export function isRevocationCheckNeeded(credential, forceRevocationCheck, revocationApi) {
+  return !!credential.credentialStatus && (forceRevocationCheck || !!revocationApi);
```

Comment:
direct de-reference here `credential.credentialStatus` is allowed because we are just checking it exists

Reply:
I'm not sure that's correct. If "credentialStatus" is redefined to mean `"http://example.com/otherMeaning"` then this will check whether `"http://example.com/otherMeaning"` property exists.

Reply:
it shouldn't matter, because if input is valid (it will be checked after this method is called) then it will verify. if not, it will fail when we expand the document. this just allows us to check if we should expand and process the document properly, without expanding the document

Reply:
Hypothetical:

```json5
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    {
      "foo": "https://www.w3.org/2018/credentials#credentialStatus"
    }
  ],
  "@type": "VerifiableCredential",
  "credentialSubject": {
    // ...
  },
  "foo": {
    // some status which, when looked up, indicates "revoked"
  }
}
```

Does/should this credential verify?

Reply:
i see your point now, in this case i think the solution requires quite some refactoring in the vc.js methods. Most methods are expecting a non-expanded credential object, and initially I thought just guarding against input we dont expect would work but I see that this should be valid but wont be. Have to also update `verifyPresentation` method to expand the presentation and use expanded credential data.